RInAppMessaging/Classes/Services/ConfigurationService.swift

```diff
@@ -54,31 +54,39 @@ internal struct ConfigurationService: ConfigurationServiceType, HttpRequestable
 
 // MARK: - HttpRequestable implementation
 extension ConfigurationService {
-
-    func buildHttpBody(with parameters: [String: Any]?) -> Result<Data, Error> {
-
+    private func getConfigRequest() throws -> GetConfigRequest {
         guard let appVersion = bundleInfo.appVersion,
             let appId = bundleInfo.applicationId,
             let sdkVersion = bundleInfo.inAppSdkVersion else {
-
                 Logger.debug("failed creating a request body")
-                return .failure(RequestError.missingMetadata)
+                throw RequestError.missingMetadata
         }
-
-        let getConfigRequest = GetConfigRequest(
+        return GetConfigRequest(
             locale: Locale.current.normalizedIdentifier,
             appVersion: appVersion,
             platform: .ios,
             appId: appId,
             sdkVersion: sdkVersion
         )
+    }
 
+    func buildURLRequest(url: URL) -> Result<URLRequest, Error> {
         do {
-            let body = try JSONEncoder().encode(getConfigRequest)
-            return .success(body)
+            let request = try getConfigRequest()
+            var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true)
+            urlComponents?.queryItems = request.toQueryItems
+            guard let url = urlComponents?.url else {
+                return .failure(RequestError.urlIsNil)
```

Comment:
Isn't it better to use existing `urlBuildingError`?

Reply:
urlBuildingError takes an error as a parameter.

Reply:
Ah ok I can see it later wrapped in it. All fine then